src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -184,3 +187,108 @@ def print_revert_report(self, *, delete_managed: bool) -> bool | None:
         return True
+
+    def _get_mapping_rules(self) -> dict[str, Rule]:
+        mapping_rules: dict[str, Rule] = {}
+        for rule in self._tm.load():
+            mapping_rules[rule.as_hms_table_key] = rule
+        return mapping_rules
+
+    def move_migrated_tables(
+        self, from_catalog: str, from_schema: str, from_table: str, to_catalog: str, to_schema: str
+    ):
+        self._ws.schemas.get(f"{from_catalog}.{from_schema}")
+        try:
+            self._ws.schemas.get(f"{to_catalog}.{to_schema}")
+        except NotFound:
+            logger.warning(f"schema {to_schema} not found in {to_catalog}, creating...")
+            self._ws.schemas.create(name=to_schema, catalog_name=to_catalog)
+
+        tables = self._ws.tables.list(catalog_name=from_catalog, schema_name=from_schema)
+        table_tasks = []
+        view_tasks = []
+        filtered_tables = [table for table in tables if from_table in [table.name, "*"]]
+        for table in filtered_tables:
+            try:
+                self._ws.tables.get(full_name=f"{to_catalog}.{to_schema}.{table.name}")
+                logger.warning(
+                    f"table {from_table} already present in {from_catalog}.{from_schema}. skipping this table..."
+                )
+                continue
+            except NotFound:
+                if table.table_type and table.table_type.value in ("EXTERNAL", "MANAGED"):
+                    table_tasks.append(
+                        partial(self._migrate_uc_table, from_catalog, from_schema, table.name, to_catalog, to_schema)
+                    )
+                else:
+                    view_tasks.append(
+                        partial(
+                            self._migrate_uc_view,
+                            from_catalog,
+                            from_schema,
+                            table.name,
+                            to_catalog,
+                            to_schema,
+                            table.view_definition,
+                        )
+                    )
+        Threads.strict(name="creating tables", tasks=table_tasks)
+        logger.info(f"migrated {len(list(table_tasks))} tables to the new schema {to_schema}.")
+        Threads.strict(name="creating views", tasks=view_tasks)
+        logger.info(f"migrated {len(list(view_tasks))} views to the new schema {to_schema}.")
+
+    def _migrate_uc_table(
+        self,
+        from_catalog: str,
+        from_schema: str,
+        from_table: str,
+        to_catalog: str,
+        to_schema: str,
+    ) -> bool:
+        from_table_name = f"{from_catalog}.{from_schema}.{from_table}"
+        to_table_name = f"{to_catalog}.{to_schema}.{from_table}"
+        try:
+            create_sql = str(next(self._backend.fetch(f"SHOW CREATE TABLE {from_table_name}"))[0])
+            create_sql = create_sql.replace(from_catalog, to_catalog).replace(from_schema, to_schema)
```

Comment:
Bug: what if some of the columns have the same name as catalog/schema? It's going to break the data